exchanges/multipart-fetch/src/multipartFetchExchange.ts

```diff
@@ -0,0 +1,196 @@
+const getOperationName = (query: DocumentNode): string | null => {
+  const node = query.definitions.find(
+    (node: any): node is OperationDefinitionNode => {
+      return node.kind === Kind.OPERATION_DEFINITION && node.name;
+    }
+  );
+
+  return node !== undefined && node.name ? node.name.value : null;
+};
+
+const createFetchSource = (operation: Operation, shouldUseGet: boolean) => {
+  if (
+    process.env.NODE_ENV !== 'production' &&
+    operation.operationName === 'subscription'
+  ) {
+    throw new Error(
+      'Received a subscription operation in the httpExchange. You are probably trying to create a subscription. Have you added a subscriptionExchange?'
+    );
+  }
+
+  return make<OperationResult>(({ next, complete }) => {
+    const abortController =
+      typeof AbortController !== 'undefined'
+        ? new AbortController()
+        : undefined;
+
+    const { context } = operation;
+    const { files } = extractFiles(operation.variables);
+
+    const extraOptions =
+      typeof context.fetchOptions === 'function'
+        ? context.fetchOptions()
+        : context.fetchOptions || {};
+
+    const operationName = getOperationName(operation.query);
+
+    const body: Body = {
+      query: print(operation.query),
+      variables: operation.variables,
+    };
+
+    if (operationName !== null) {
+      body.operationName = operationName;
+    }
+
+    const fetchOptions = {
+      ...extraOptions,
+      method: shouldUseGet ? 'GET' : 'POST',
+      headers: {
+        'content-type': 'application/json',
+        ...extraOptions.headers,
+      },
+      signal:
+        abortController !== undefined ? abortController.signal : undefined,
+    };
+
+    if (!!files.size) {
+      fetchOptions.body = new FormData();
+      fetchOptions.method = 'POST';
+      // Make fetch auto-append this for correctness
+      delete fetchOptions.headers['content-type'];
+
+      fetchOptions.body.append('operations', JSON.stringify(body));
+
+      const map = {};
+      let i = 0;
+      files.forEach(paths => {
+        map[++i] = paths.map(path => `variables.${path}`);
+      });
+      fetchOptions.body.append('map', JSON.stringify(map));
+
+      i = 0;
+      files.forEach((_, file) => {
+        (fetchOptions.body as FormData).append(`${++i}`, file, file.name);
+      });
+    } else if (shouldUseGet) {
+      operation.context.url = convertToGet(operation.context.url, body);
+    } else {
+      fetchOptions.body = JSON.stringify(body);
+    }
+
+    let ended = false;
+
+    Promise.resolve()
+      .then(() => (ended ? undefined : executeFetch(operation, fetchOptions)))
+      .then((result: OperationResult | undefined) => {
+        if (!ended) {
+          ended = true;
+          if (result) next(result);
+          complete();
+        }
+      });
+
+    return () => {
+      ended = true;
+      if (abortController !== undefined) {
+        abortController.abort();
+      }
+    };
+  });
+};
+
+const executeFetch = (
```

Comment:
My thinking was, in case we need to seperate these two more we're already prepared, else we would also introduce another export from `@urql/core` which increases the size